app/javascript/packs/admin_invoices/body.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import AlertContainer from 'react-alert';
+import Fetch from '../Fetch';
+import Content from './content';
+
+class AdminInvoices extends React.Component {
+  constructor(props) {
+    super(props);
+    const userIdPattern = /users\/\d+/g;
+    this.state = {
+      invoices: [],
+      user: [],
+      salaryType: '',
+      salary: 0,
+      dayOffs: 0,
+      userId: window.location.href.match(userIdPattern)[0].replace('users/', ''),
+    };
+    this.loadInvoices = this.loadInvoices.bind(this);
+    this.handleDateChange = this.handleDateChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadInvoices(new Date());
+  }
+
+  loadInvoices(date) {
+    Fetch.json(`/api/v1/admin/user/users/${this.state.userId}/invoices`, {
+      date: date,
+    })
+      .then((data) => {
+        this.setState({
+          invoices: data[0],
+          user: data[1],
+          salaryType: data[2],
+          salary: data[3],
+          dayOffs: data[4],
+        });
+      });
+  }
+
+  handleDateChange(event) {
+    this.loadInvoices(event.target.value.concat('-01'));
+  }
+
+
+  render() {
+    const today = new Date();
+    const year = today.getUTCFullYear().toString();
+    let month = (today.getUTCMonth() + 1).toString();
+    if (month.length === 1) {
+      month = '0'.concat(month);
```

Comment:
I think it is a good time to introduce `moment.js` which can simplify our code. We have enough places where we work with date/time.
But it is up to you.